src/hp/key/node.rs

```diff
@@ -30,6 +30,12 @@ impl Hash for PublicKey {
     }
 }
 
+impl std::cmp::PartialEq for PublicKey {
```

Comment:
why this manual impl?

Reply:
this was the clippy rule we were getting warned about:
https://rust-lang.github.io/rust-clippy/master/index.html#derive_hash_xor_eq